src/store/connectionStore.ts

```diff
@@ -78,29 +102,67 @@ export const useConnectionStore = defineStore('connectionStore', {
   },
   actions: {
     async fetchConnections() {
-      this.connections = (await storeApi.get('connections', [])) as Connection[];
+      try {
+        const connections = await storeApi.get('connections', []) as Connection[];
+        this.connections = connections.map(conn => {
+          if ('host' in conn && 'port' in conn) {
+            return { ...conn, type: DatabaseType.ELASTICSEARCH };
+          } else if ('region' in conn && 'accessKeyId' in conn) {
+            return { ...conn, type: DatabaseType.DYNAMODB };
+          }
+          return conn;
+        });
+      } catch (error) {
+        console.error('Error fetching connections:', error);
+        this.connections = [];
+      }
     },
-    async testConnection(con: Connection) {
+    async testConnection(con: ElasticsearchConnection) {
       const client = loadHttpClient(con);
 
       return await client.get(con.indexName ?? undefined, 'format=json');
     },
-    async saveConnection(connection: Connection) {
-      const index = this.connections.findIndex(({ id }: Connection) => id === connection.id);
-      if (index >= 0) {
-        this.connections[index] = connection;
-      } else {
-        this.connections.push({ ...connection, id: this.connections.length + 1 });
+    async saveConnection(connection:Connection) {
+      try {
+        if (connection.id) {
+          const index = this.connections.findIndex(c => c.id === connection.id);
+          if (index !== -1) {
+            this.connections[index] = connection;
+          }
+        } else {
+          connection.id = this.connections.length + 1;
+          this.connections.push(connection);
+        }
+        
+        try {
+          await storeApi.set('connections', pureObject(this.connections));
+        } catch (error) {
+          console.warn('Failed to persist connections:', error);
+        }
+        
+        return connection;
+      } catch (error) {
+        console.error('Error saving connection:', error);
+        throw error;
```

Comment:
store the type to type like DatabaseType.ELASTICSEARCH to storeApi.set, it should help to simplify the fetchConnections as well?